core/src/test/java/org/wso2/msf4j/util/client/websocket/WebSocketClient.java

```diff
@@ -177,12 +177,12 @@ public void sendBinary(ByteBuffer buf) {
      *
      * @param buf content of the pong message to be sent.
      */
-    public void sendPong(ByteBuffer buf) {
+    public void sendPing(ByteBuffer buf) {
         if (channel == null) {
             log.error("Channel is null, cannot write data.");
             throw new NullPointerException("Cannot find the channel to write");
         }
-        channel.writeAndFlush(new PongWebSocketFrame(Unpooled.wrappedBuffer(buf)));
+        channel.writeAndFlush(new PingWebSocketFrame(Unpooled.wrappedBuffer(buf)));
```

Comment:
The way it was implemented previously was wrong. If we send a ping message other side should send a pong. There was a bug where we sent a pong and received a pong. Now that issue is fixed in carbon-transports. So when we send a ping to WebSocket server it is automatically sending a pong message to the WebSocket client.